bazel/integration/test_runner/package_json.ts

```diff
@@ -82,6 +106,24 @@ function updateMappingForRecord(record: DependencyRecord, mappings: PackageMappi
       continue;
     }
 
-    record[pkgName] = mappedAbsolutePath;
+    record[entryKey] = mappedAbsolutePath;
+  }
+}
+
+/**
+ * Gets the package name from a dependency record entry.
+ *
+ * @param entryKey Key of the dependency record entry.
+ * @param nameMatchRegex Optional regular expression that can be specified to match the package
+ *   name in a dependency entry. This is useful for e.g. Yarn resolutions using patterns.
+ */
+function getPackageNameFromDependencyEntry(
+  entryKey: string,
+  nameMatchRegex?: RegExp,
+): string | null {
+  if (nameMatchRegex) {
+    const matches = entryKey.match(nameMatchRegex);
+    return matches === null ? null : matches[1];
```

Comment:
Nit: This could be simplified as:

```suggestion
    return matches?.[1];
```

Reply:
good point. this would return `undefined` through, unless we use an `??` later though. my preference has been to return `null` in explicit cases, so I'll keep it as is, but please let me know if you feel differently about `undefined vs. null`